tuna/rocmlir/rocmlir_tables.py

```diff
@@ -0,0 +1,269 @@
+  direction = Column(String(length=8), nullable=False)
+  in_channels = Column(Integer, nullable=False, server_default="0")
+  in_h = Column(Integer, nullable=False, server_default="0")
+  in_w = Column(Integer, nullable=False, server_default="0")
+  fil_h = Column(Integer, nullable=False, server_default="0")
+  fil_w = Column(Integer, nullable=False, server_default="0")
+  out_channels = Column(Integer, nullable=False, server_default="0")
+  batchsize = Column(Integer, nullable=False, server_default="0")
+  pad_h = Column(Integer, nullable=False, server_default="0")
+  pad_w = Column(Integer, nullable=False, server_default="0")
+  conv_stride_h = Column(Integer, nullable=False, server_default="0")
+  conv_stride_w = Column(Integer, nullable=False, server_default="0")
+  dilation_h = Column(Integer, nullable=False, server_default="0")
+  dilation_w = Column(Integer, nullable=False, server_default="0")
+  group_size = Column(Integer, nullable=False, server_default="0")
+  kernel_repeats = Column(Integer, nullable=False, server_default="0")
+
+  def __repr__(self) -> str:
+    return f"ConvolutionConfig {self.to_dict()}"
+
+  options = {
+    'direction': '-F',
+    'fil_layout': '-f',
+    'in_layout': '-I',
+    'out_layout': '-O',
+    'batchsize': '-n',
+    'in_channels': '-c',
+    'in_h': '-H',
+    'in_w': '-W',
+    'out_channels': '-k',
+    'fil_h': '-y',
+    'fil_w': '-x',
+    'pad_h': '-p',
+    'pad_w': '-q',
+    'conv_stride_h': '-u',
+    'conv_stride_w': '-v',
+    'dilation_h': '-l',
+    'dilation_w': '-j',
+    'group_size': '-g',
+    'data_type': '-t',
+# getopt in ConvConfiguration.fromCommandLine only does single-char options.
+# Count on tuneMLIRKernels to set config.MLIR_N_REPEATS to 1.
+#    'kernel_repeats': '--kernel-repeats',
+    'kernel_repeats': None,
+    'id': None,
+    'valid': None
+  }
+
+  def config_string(self):
+    """Return config as a flag/value string suitable for tuningRunner.py."""
+    string = "conv "                    # +++pf:  of course generalise for gemm
+    for field, value in self.to_dict().items():
+      flag = self.options[field]
+      if flag:
+        string += f"{flag} {value} "
+    string += "-m conv"
+    return string
+
+
+class ConvolutionResults(BASE):  # pylint: disable=too-many-instance-attributes
+  """Collects the results of convolution tuning.
+  """
+  __tablename__ = "rocmlir_conv_results"
+  __table_args__ = (UniqueConstraint("config", "session", name="uq_idx"),)
+
+  @orm.reconstructor
+  def __init__(self, **kwargs):
+    if 'logger' in kwargs:
+      self.logger = kwargs['logger']
+    else:
+      self.logger = setup_logger('results')
+
+  @declared_attr
+  def session(self):
+    """session column"""
+    return Column(Integer, ForeignKey("session_rocmlir.id"), nullable=False)
+
+  config = Column(Integer, ForeignKey("rocmlir_conv_config.id"), nullable=False)
+  config_str = Column(Text, nullable=False)
```

Comment:
The config_str field may go away again.  I added it because I get a string from the runner and needed to keep it, but then I realised that the job has the config ID and I could save that and reconstruct the string when exporting the results table.